doc/device_client.md

```diff
@@ -0,0 +1,32 @@
+# Azure IoT Device Client Library
+
+The Azure IoT Device Client Library provides an efficient and well tested API for connecting and using devices with the Azure IoT Hub.
+
+## client.setOptions(options, [callback])
+
+Used directly after creating the Device Client to set options on the client. These should not be dynamically altered during runtime of the device client.
+
+### Available Options
+
+- `ca`: Public certificate in PEM form for certificate authority being used by the Hub service.  This is the CA that the hub is using to secure TLS connections and the client validates the connection using this public cert in order to validate the identity of the hub.  If you are connecting to an Azure IoT Hub inside of an Azure data center, you do not need to set this.  If you are connecting to some other hub (e.g. an Edge Hub), then you may need to set this to the server cert that the hub uses for TLS.
+- `keepalive`: Keepalive interval in numeric format (seconds). This controls the keepalive ping for MQTT specifically. If you are using AMQP or HTTP, this will do nothing.
+- `productInfo`: Custom user defined information to be appended to existing User Agent information. The User Agent Identification information is used predominantly by Microsoft internally for identifying metadata related to Device Client usage for Azure IoT.
+- `modelId`: !!Digital Twin Use Only!! String used in MQTT username setting the Digital Twin modelId.
+- `tokenRenewal`: Optional object with token renewal values.  Only use with authentication that uses pre shared keys.
+- `mqtt`: Options specific to the MQTT transport
+    - `webSocketAgent`: [Agent]{@link https://nodejs.org/api/https.html#https_class_https_agent} object to use with MQTT-WS connections. Can be used for tunneling proxies.
```

Comment:
in the case of protocol specific options, do I need to specify the protocol? Or will a call to client.SetOptions(webSocketAgent=xxx) automatically work for whichever protocol I'm using (assuming amqp or mqtt in this case based on the parameter name)

Reply:
This is incorrect. what's a good way to guide users that they need to put it in an object?

i.e.: `client.setOptions({mqtt: {websocketAgent: agent}}`

(also, I know the API looks kinda dumb, since all three can accept an agent, but,,, that's just how it is and we can't break it now).

Reply:
yeah it's fine, doesn't have to be perfect as long as it's usable-- in this case, probably best to just show an example. As in, before starting the protocol specific examples with `mqtt`, a line saying:

 **Protocol specific options**: to use options for a specific protocol, you must pass it as an object that specifies both the protocol used and the option you are setting. For example, `client.setOptions({mqtt: {websocketAgent: agent}})`.